### PR TITLE
docs: initial auth0 guide

### PR DESCRIPTION
**Pull request by tilschuenemann**
This PR adds a guide for people using Auth0 who want to migrate to Ory, featuring:

1. exporting hashed passwords from Auth0
2. creating a bulk user export for getting user data
3. creating identites

## Checklist

- [x] I have read the [contributing guidelines](../blob/master/CONTRIBUTING.md) and signed the CLA.
- [x] I have referenced an issue containing the design document if my change introduces a new feature.
- [x] I have read the [security policy](../security/policy).
- [x] I confirm that this pull request does not address a security vulnerability. 
      If this pull request addresses a security vulnerability, 
      I confirm that I got approval (please contact [security@ory.sh](mailto:security@ory.sh)) from the maintainers to push the changes.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] I have added the necessary documentation within the code base (if appropriate).

### CODE REVIEW

**Comment by tilschuenemann** [comment]
I still need to to:
* make sure to conform to our doc guidelines
* rewrite the start / ending section
* add some more references to the ending to other parts of the docs for further configuration options (ie. enable password reset flow) 

There are some minor hiccups where I am not sure how to proceed:
* small wording issues, eg. creating identities for Ory Kratos / Network?
* Is there a way to create a PAT using the CLI or can I create identities without using one?
* Where should we host the scripts, so that the reader can just download them with curl? Or should we leave them in the guide like this?

**Comment by vinckr** [comment]
hiccups:
- Let me know when it is ready for review and I will go over the text and see if I can fix any naming issues.
- There should be a guide how to use/create the PAT, just link to that.
- we can host the scripts here, maybe at https://github.com/ory/docs/tree/master/code-examples in a /scripts folder depending how many there are.

File: docs/migrate-to-ory/auth0.mdx [hunk 1]
```diff
@@ -0,0 +1,263 @@
...
+        )
+    fi
+}
+
+create_identity() {
+    if [[ -z "$payload" ]]; then
+        echo "please supply a valid payload"
+        exit 1
+    else
+        curl --request POST -sL \
```
Comment (aeneasr) [inline docs/migrate-to-ory/auth0.mdx:224]: Consider using `ory import identities` instead, it allows importing many identities from a JSON source

**Review by aeneasr — COMMENTED** [review]
This is a great start :) We'll need to docsify this a bit (I'm sure @tomekpapiernik can help here :) ) and then this should already be good to go for the first step. Adding how to migrate the SDKs etc we can to at a later point

File: docs/migrate-to-ory/auth0.mdx [hunk 2]
```diff
@@ -0,0 +1,263 @@
...
+```bash
+export AUTH0_DOMAIN="{your_auth0_domain}.auth0.com"
+export AUTH0_CONNECTION_ID="{your_auth0_connection_id}"
+export AUTH0_TOKEN="{your_auth0_management_api_token}"
+```
+
+> Put this online, use curl and run this?
+
+```bash
+job_response=$(
```
Comment (aeneasr) [inline docs/migrate-to-ory/auth0.mdx:95]:
```suggestion
#!/bin/bash

job_response=$(
```

File: docs/migrate-to-ory/auth0.mdx [hunk 3]
```diff
@@ -0,0 +1,263 @@
...
+export ORY_SLUG='{your_ory_slug}'
+export ORY_TOKEN='{ory_token}'
+export AUTH0_USERDATA="{/path/to/AUTH0_USERDATA.json}"
+export AUTH0_PWEXPORT="{/path/to/hashed_passwords.json}" # comment this line to reserve not create identities
+```
+
+Now we can create the identities using this script.
+
+```bash
+create_payload() {
```
Comment (aeneasr) [inline docs/migrate-to-ory/auth0.mdx:190]:
```suggestion
#!/bin/bash

create_payload() {
```

**Comment by tomekpapiernik** [comment]
Hey @tilschuenemann , before you hand it over for review, please make sure you follow the guidelines: 

https://github.com/ory/docs#style-guide

The thing that's the most important is to move away from the "we" form. The right way to present instruction is to write in present tense, imperative, and refer to the audience directly ("you" - "you must do this to get this result").

Ping me when you're ready for a proper review :)

**Comment by tilschuenemann** [comment]
* the guide should conform to our doc guidelines now
* the changes for the scripts are addressed and they reside in their own files in `code-examples/migrate-to-ory`. 
    * before merging to master we have to change the URLs in the guide to refer to the correct branch!
* ory import identities is used now

File: docs/migrate-to-ory/auth0.mdx [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+2. run a bulk user export to get your complete user data.
+
+3. create identities in Ory Network.
+
+### Export password hashes
+
+Because password hashes are considered sensitive information, Auth0 doesn't export them as part of the general export process. You
+must file a support ticket to get the hashes and other password related information.
+
+! **When you are exporting hashed passwords, neither you, nor Auth0 or the Ory system has access to the plain text password** !
```
Comment (tomekpapiernik) [inline docs/migrate-to-ory/auth0.mdx:27]: Please use Docusaurus admonitions for notes like this one: 

https://docusaurus.io/docs/markdown-features/admonitions

You can find many examples of these in our docs. ;)

Comment (tomekpapiernik) [reply docs/migrate-to-ory/auth0.mdx:27]: I won't add this comment to every applicable place, just make sure to do it for every note-thingy.

File: docs/migrate-to-ory/auth0.mdx [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+When you have exported the hashed passwords from Auth0 and imported them into Ory Kratos, your users will be able to log in with
+the same password they used previously.
+
+**Tip:**
+
+> If you aren't able to export the hashes, due to the plan you are on or the migration timelines, one alternative is to have every
+> user change their password using the forgot password flow. You can still import all the user data.
+
+To start the process of obtaining the hashes, open a support ticket. From your dashboard, navigate to Get Support, and then open a
+ticket by scrolling to the bottom of the support page.
```
Comment (tomekpapiernik) [inline docs/migrate-to-ory/auth0.mdx:38]: Use the right formatting when talking about UI elements. Refer to the guidelines for more info.

File: docs/migrate-to-ory/auth0.mdx [hunk 6]
```diff
@@ -0,0 +1,140 @@
...
+
+![Image of Auth0 Support Page](_static/1.png)
+
+Select the issue and details of your request. Choose `I have a question regarding my Auth0 account` and then pick the
+`I would like to obtain an export of my password hashes` option.
```
Comment (tomekpapiernik) [inline docs/migrate-to-ory/auth0.mdx:43]: UI elements formatting

File: docs/migrate-to-ory/auth0.mdx [hunk 7]
```diff
@@ -0,0 +1,140 @@
...
+```
+
+```shell
+bash <(curl https://raw.githubusercontent.com/ory/docs/auth0guide/code-examples/migrate-to-ory/0-get-auth0-user-data.sh)
+```
+
+### Creating identities
+
+Once the download is finished and Auth0 provided you with the hashed passwords, you can create the identities
+`in Ory Kratos / Ory Network?`, requiring the users to login with email and password.
```
Comment (tomekpapiernik) [inline docs/migrate-to-ory/auth0.mdx:97]: Clean up these comments, please. There are a few of those still in the doc.

File: docs/migrate-to-ory/auth0.mdx [hunk 8]
```diff
@@ -0,0 +1,140 @@
...
+
+Since it's possible your user data contains new users who weren't registered when the hashed passwords were exported, we'll
+reserve an identity for that email. If you're on a free Auth0 plan and you don't qualify for an export of your hashed passwords,
+then this might be a viable option for you (`pw reset flow?`).
+
+Now you create an Ory Project incase you don't have one yet:
+
+`setting schema this isnt pretty - i know, see https://github.com/ory/docs/pull/1157`
```
Comment (tomekpapiernik) [inline docs/migrate-to-ory/auth0.mdx:105]: 🙄

File: docs/migrate-to-ory/auth0.mdx [hunk 9]
```diff
@@ -0,0 +1,140 @@
...
+```
+
+Export the paths to your hashed passwords and your Auth0 user data like this:
+
+```shell
+export AUTH0_USERDATA="{/path/to/AUTH0_USERDATA.json}"
+export AUTH0_PWEXPORT="{/path/to/hashed_passwords.json}" # comment this line to reserve not create identities
```
Comment (tomekpapiernik) [inline docs/migrate-to-ory/auth0.mdx:121]: `comment this line to reserve not create identities` 
???

File: docs/migrate-to-ory/auth0.mdx [hunk 10]
```diff
@@ -0,0 +1,140 @@
...
+`I would like to obtain an export of my password hashes` option.
+
+You’ll receive an automated message after this ticket is submitted. View progress on the ticket screen.
+
+**Important!**
+
+> Auth0 requires you to be on a paid plan to export your password hashes. You cannot export them otherwise. Additionally, Auth0
+> makes no commitments about the processing time for a password hash request. Plan accordingly.
+
+After your request is processed, you’ll download a gzipped JSON file containing password hashes, related information and user ids.
```
Comment (tomekpapiernik) [inline docs/migrate-to-ory/auth0.mdx:52]: "gzipped"? How about "compressed JSON file"?

Comment (tomekpapiernik) [inline docs/migrate-to-ory/auth0.mdx:52]: ids -> IDs

**Review by tomekpapiernik — CHANGES_REQUESTED** [review]
Please fix the highlighted issues before we proceed. 
On an additional note - the Auth0 screenshot doesn't seem to be very helpful.

File: docs/migrate-to-ory/auth0.mdx (hunk 10 above)
Comment (tilschuenemann) [reply docs/migrate-to-ory/auth0.mdx]: That's better!

File: docs/migrate-to-ory/auth0.mdx (hunk 9 above)
Comment (tilschuenemann) [reply docs/migrate-to-ory/auth0.mdx]: I've added an extra option to make it less confusing

**Comment by tomekpapiernik** [comment]
@tilschuenemann I have just pushed an edited version of the document. Please look it over with someone to make sure nothing was "lost in translation" ;). 

What's up with that empty file?  -> `docs/kratos/manage-identities/identity-schema/presets.mdx` ?
Also, aren't there a few files too many in this PR?

**Comment by tilschuenemann** [comment]
@tomekpapiernik It's alot better now, thank you! I've removed the unnecessary files and made some small adjustments.
When we're ready to merge I'll adjust the branch of the scripts  URL to master and then we're good to go from my side :)

**Review by tilschuenemann — COMMENTED** [review]
Should all be fine now.

**Review by tomekpapiernik — APPROVED** [review]
Looks good now, merging!

**Pull request merged**